### PR TITLE
tools: Replace string functions with secure safe string

### PR DESCRIPTION
**Pull request by anandaravuri**
Changes:
   - Replace strlen() with strlen_s()
   - Replace snprintf () with snprintf_s_X ()

### CODE REVIEW

File: tools/fpgaconf/fpgaconf.c [hunk 1]
```diff
@@ -273,7 +273,7 @@ int parse_args(int argc, char *argv[])
 				break;
 			endptr = NULL;
 			config.target.bus = (int) strtoul(tmp_optarg, &endptr, 0);
-			if (endptr != tmp_optarg + strlen(tmp_optarg)) {
+			if (endptr != tmp_optarg + strnlen_s(tmp_optarg, sizeof(tmp_optarg))) {
```
Comment (luebbers) [inline tools/fpgaconf/fpgaconf.c]: How about the following faster alternative that doesn't need `strlen()` at all:

    if (*tmp_optarg == '\0' || *endptr != '\0') {
        fprintf(...)
        ...

This would treat the bus argument as invalid if it's either empty (`tmp_optarg` points to '\0') or invalid (`endptr` points to something other that a '\0').

Actually, the existing code would accept an empty argument and set the bus to `0`, so

    if (*endptr != '\0') {
        fprintf(...)
        ...

would be enough. The check for an empty argument doesn't hurt, though.

File: tools/fpgaconf/fpgaconf.c [hunk 2]
```diff
@@ -306,7 +306,7 @@ int parse_args(int argc, char *argv[])
 				break;
 			endptr = NULL;
 			config.target.socket = (int) strtoul(tmp_optarg, &endptr, 0);
-			if (endptr != tmp_optarg + strlen(tmp_optarg)) {
+			if (endptr != tmp_optarg + strnlen_s(tmp_optarg, sizeof(tmp_optarg))) {
```
Comment (luebbers) [inline tools/fpgaconf/fpgaconf.c]: Same as above for all changes.

File: tools/ras/main.c [hunk 3]
```diff
@@ -1067,7 +1067,7 @@ fpga_result mmio_error(struct RASCommandLine *rasCmdLine)
 	if ( rasCmdLine->function >0 )
 		function = rasCmdLine->bus;
 
-	snprintf(sysfs_path, sizeof(sysfs_path),
+	snprintf_s_iiii(sysfs_path, sizeof(sysfs_path),
```
Comment (luebbers) [inline tools/ras/main.c]: Can you explain why is this necessary? Is `snprintf()` with four integer arguments unsafe?

Comment (luebbers) [reply tools/ras/main.c]: I see.

In this case the format string is a constant containing four `%i`, and the arguments are either constants (`0`) or `int` - is `sprintf_iiii()` necessary in this case?

Comment (tswhison) [reply tools/ras/main.c]: I think so, or else we could try to get a per-instance waiver; even then, we would still have to deal with the issue when Klocwork flags it.

**Comment by anandaravuri** [comment]
Not need
